roadmap/roadmap.adoc

```diff
@@ -0,0 +1,158 @@
+include::../shared/issue-footnotes.adoc[]
+
+:m3: ../metametamodel/metametamodel
+:refarch: ../reference-architecture/reference-architecture
+:serialization: ../serialization/serialization
+
+= LionWeb Roadmap, Releases, and History
+:toc:
+:toclevels: 4
+// :sectnums:
+// :sectnumlevels: 4
+
+== Roadmap
+As a volunteer-run Open Source effort, we cannot commit to a timeline.
+We're actively looking for corporate sponsors, with some prospects in sight.
+
+=== In progress
+
+[#bulk-protocol]
+==== Specifying bulk protocol
+We've decided{fn-org148} to specify protocols only by their conceptual features.
+The first such protocol will be the https://github.com/LionWeb-org/organization/blob/d8a54813c6bd224ef84e53a26ac7d60a45a866b5/bulk/repo-access-api.adoc[Bulk Repository access API].
+It is used to store and retrieve nodes in batches at the moment of invocation.{fn-org25}
+It is intended for CRUD operations on (larger) sets of nodes.
+Additionally, this API allows to list the available <<{m3}.adoc#partition, Partitions>>, get Languages known to the repository, and request unused <<{m3}.adoc#identifiers, node ids>>.
+
+=== Planned
+We highly encourage any participation in LionWeb.
+So if you'd like to work on some of these features, or even only have an urgent need for them: let us know!
+This will influence the feature's priority.
+
+The list below is in no particular order.
+
+[#delta-protocol]
+==== Delta protocol
+As described in our <<{refarch}.adoc#delta, reference architecture>>, via delta protocol a client receives an initial snapshot similar to the bulk case.
+Afterwards, the client continues to receive changes (aka deltas) from the repo; it also writes back changes continuously.{fn-org28}
+
+Open questions for this protocol include
+
+* Which messages do we support?
+For example, is _move a child from position 2 to position 5 in the same list_ the same as _move a child from parent A to parent B_ the same message kind?
+
+* Is the protocol the same for client -> repository updates vs. repository -> client updates?
+That's especially relevant for (web-based) editors.
+
+* How strong do we couple the protocol to an implementation?
+Both https://en.wikipedia.org/wiki/Operational_transformation[operational transformation] and https://en.wikipedia.org/wiki/Conflict-free_replicated_data_type[CRDTs] can be used to implement consistent clients.
+For both of them, several implementations are available.
+Additionally, we might support simpler clients that assume no conflicts or missing updates at all.
```

Comment:
And: Can we find a sensible common denominator to both implementation (OT, CRDT)?

Reply:
In line with Federico's comment https://github.com/LionWeb-org/organization/pull/174#discussion_r1336097035, I'm moving the details to #28